cli/src/cmd/cast/run.rs

```diff
@@ -103,7 +103,7 @@ impl RunArgs {
         // configures a bare version of the evm executor: no cheatcode inspector is enabled,
         // tracing will be enabled only for the targeted transaction
         let builder =
-            ExecutorBuilder::default().with_config(env).with_spec(evm_spec(&config.evm_version));
+            ExecutorBuilder::default().with_config(env).with_spec(SpecId::SHANGHAI);
```

Comment:
hmm, not sure about this because this bypasses the config

Reply:
Hmm this is interesting, because the push0 can come from the user's code OR a forked network's code. 

If building something multichain, they might have their EVM config set to paris so it can work on mainnet and optimism. They may debug a test/tx against mainnet which has a push0, and because their code will be deployed to mainnet (shanghai), that debug experience should work. Not changing this means it would fail, because live code had a push0. Which means we SHOULD change the EVM version to shanghai for that case, even locally they chose paris.

Since we can't know the user's intentions (about what chains they'll use), I think we have to change this to shanghai to support the above case? And then maybe we just show a warning explaining that it was changed and why it was changed

Reply:
The reasoning for this change is that since this is only for the debugger, shanghai enables push0 code but doesnt disable other codes